### PR TITLE
1469575: purge orphanded pools - 2.0

### PR DESCRIPTION
**Pull request by vritant**

### CODE REVIEW

**Comment by vritant** [comment]
[Investigate DB script that can be run by liquibase to clean up rogue pools.](https://trello.com/c/uKtpqwWt/487-investigate-db-script-that-can-be-run-by-liquibase-to-clean-up-rogue-pools)

File: server/src/main/resources/db/changelog/20170718115416-purge-orphan-stack-derived-pools.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            where id not in
+            (    select pool_id
+                 from cp2_pool_source_sub
+            )
+              and type = 'BONUS';
+        </sql>
+    </changeSet>
+
+</databaseChangeLog>
+<!-- vim: set expandtab sts=4 sw=4 ai: -->
```
Comment (mstead) [inline server/src/main/resources/db/changelog/20170718115416-purge-orphan-stack-derived-pools.xml]: This file looks to have been added in error. Please remove.

File: server/src/main/resources/db/changelog/20170718115416-purge-orphaned-pools.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        </sql>
+        <sql>
+            delete from
+            cp_pool
+            where id not in
+            (    select pool_id
+                 from cp2_pool_source_sub
+            )
+              and type = 'BONUS';
+        </sql>
```
Comment (mstead) [inline server/src/main/resources/db/changelog/20170718115416-purge-orphaned-pools.xml]: Will not run on MySQL.

File: server/src/main/resources/db/changelog/20170718115416-purge-orphaned-pools.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    <changeSet id="20170718115416-1" author="vrjain">
+        <comment> purge orphaned pools</comment>
+        <sql>
+            delete from cp_pool where id in
+            (  select p.id
+               from cp_pool p
+                   left outer join cp_pool_source_stack ss
+               on p.id = ss.derivedpool_id
+               where p.type='STACK_DERIVED'
+                 and ss.derivedpool_id is null
```
Comment (Ceiu) [inline server/src/main/resources/db/changelog/20170718115416-purge-orphaned-pools.xml]: This won't actually match anything.

**Comment by mstead** [comment]
After speaking with mccune, we decided to go with a script + Kbase instead of an automatic cleanup due to complexity and added risk. Closing.

**Pull request closed without merging**